File: apps/api/routes/pdf.py
from fastapi import APIRouter
from controllers.pdf import PDFController

router = APIRouter()
pdf_controller = PDFController()

@router.get("/pdf/ans")
async def ans_pdf_scrapper(target_url: str):
    return pdf_controller.ans_pdf_scrapper(target_url)

@router.get("/pdf/scrap")
async def get_ans_pdf(target_url: str, target_file: str):
    return pdf_controller.get_ans_pdf(target_url, target_file)

File: apps/frontend/src/services/pdf.ts
import { API } from '@/libs/api';

export async function getANSZippedPDFs({ url }: { url: string }) {
    const params = {
        target_url: url,
    };

    const { data } = await API.get('/pdf/ans', { params });

    return data;
}

export async function getPDF({ url, filename }: { url: string; filename: string }) {
    const params = {
        target_url: url,
        target_file: filename,
    };

    const { data } = await API.get('/pdf/scrap', { params });

    return data;
}
